akka-stream/src/main/scala/akka/stream/impl/fusing/ActorGraphInterpreter.scala

```diff
@@ -605,6 +605,14 @@ import scala.util.control.NonFatal
     // call has no effect and therefore does the right thing: nothing.
     try {
       inputs.foreach(_.onInternalError(reason))
+      // fail all sources
+      logics.foreach { logic ⇒
+        // source
+        if (logic.inCount == 0 && logic.outCount > 0) {
```

Comment:
What's the rationale for this condition? Why not just run it for all stages which are not yet finished?